packages/react-router/lib/dom/ssr/routeModules.ts

```diff
@@ -18,19 +18,82 @@ export interface RouteModules {
   [routeId: string]: RouteModule | undefined;
 }
 
+/**
+ * The shape of a route module shipped to the client
+ */
 export interface RouteModule {
+  /**
+   * Action function called only in the browser.  Route client loaders provide
+   * data to route components in addition to, or in place of, route loaders.
+   */
   clientAction?: ClientActionFunction;
+  /**
+   * Like route actions but only called in the browser.
+   */
   clientLoader?: ClientLoaderFunction;
-  ErrorBoundary?: ErrorBoundaryComponent;
-  HydrateFallback?: HydrateFallbackComponent;
+  /**
+   * ErrorBoundary to display for this route
+   */
+  ErrorBoundary?: React.ComponentType;
+  /**
+   * `<Route HydrateFallback>` component to render on initial loads
+   * when client loaders are present
+   */
+  HydrateFallback?: React.ComponentType;
+  /**
+   * The `Layout` export is only applicable to the root route.  Because the root
+   * route manages the document for all routes, it supports an additional optional
+   * `Layout` export that will be wrapped around the rendered UI, which may come
+   * from the default `Component` export, the `ErrorBoundary`, or the `HydrateFallback`
+   */
   Layout?: LayoutComponent;
-  default: RouteComponent;
+  /**
+   * Defines the component that will render when the route matches.
+   */
+  default: React.ComponentType<{}>;
```

Comment:
Inlined but I think this is invalid now with the typegen props - going to look into making this actually pull the typegen stuff (`CreateComponentProps`) with `unknown` values for route-specific info